### ISSUE TITLE
#TODO

### ISSUE THREAD

**Issue by nateanl** [speechLabBcCuny/onssen#2]
Current result:
![image](https://user-images.githubusercontent.com/8653221/46262099-a0804980-c4ca-11e8-807a-40c190f249be.png)

There are still many things to improve. 
- The result of mask inference head is very bad.  Fix it. [High Priority]
- Evaluate the current result. [High Priority]
- Batch training. The current training code calls the backward function at every batch. Theoretically, the function is called only once an epoch, it saves the loss for each batch and backward at the end. This may cause the memory explosion. [Medium Priority]
- Replace the batch generator code with "DataLoader" in Pytorch. [Medium Priority]
- Use GPU. [Low Priority]
